### PR TITLE
Add RPC HTTP options to specify custom truststore and password

### PR DESCRIPTION
**Pull request by pullurib**
## PR description

- Introduce a new option to specify the truststore for the JSON-RPC HTTP client.
- Support passing a password file for the truststore through a new option to avoid exposing the password in logs or the CLI.
- Ensure the client certificates are validated successfully using the provided truststore - added test cases 
- Added testcase to verify option use 

## Fixed Issue(s)
#7932

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -44,6 +44,7 @@
 - Add account and state overrides to `eth_call` [#7801](https://github.com/hyperledger/besu/pull/7801) and `eth_estimateGas` [#7890](https://github.com/hyperledger/besu/pull/7890)
 - Prometheus Java Metrics library upgraded to version 1.3.3 [#7880](https://github.com/hyperledger/besu/pull/7880)
 - Add histogram to Prometheus metrics system [#7944](https://github.com/hyperledger/besu/pull/7944)
+- Add RPC HTTP options to specify custom truststore and it's password [#7978](https://github.com/hyperledger/besu/pull/7978)
```
Comment (macfarla) [inline CHANGELOG.md:47]:
```suggestion
- Add RPC HTTP options to specify custom truststore and its password [#7978](https://github.com/hyperledger/besu/pull/7978)
```

File: besu/src/main/java/org/hyperledger/besu/cli/options/JsonRpcHttpOptions.java [hunk 2]
```diff
@@ -394,10 +411,11 @@ private void validateTls(final CommandLine commandLine) {
 
     if (isRpcHttpTlsClientAuthEnabled
         && !isRpcHttpTlsCAClientsEnabled
-        && rpcHttpTlsKnownClientsFile == null) {
+        && rpcHttpTlsKnownClientsFile == null
+        && rpcHttpTlsTruststoreFile == null) {
       throw new CommandLine.ParameterException(
           commandLine,
-          "Known-clients file must be specified or CA clients must be enabled when TLS client authentication is enabled for JSON-RPC HTTP endpoint");
+          "Known-clients file or Truststore file must be specified or CA clients must be enabled when TLS client authentication is enabled for JSON-RPC HTTP endpoint");
```
Comment (macfarla) [inline besu/src/main/java/org/hyperledger/besu/cli/options/JsonRpcHttpOptions.java:418]: what happens if both are specified? should it be one or the other?

Comment (pullurib) [reply besu/src/main/java/org/hyperledger/besu/cli/options/JsonRpcHttpOptions.java]: Yeah, it should be one or the other . Added option dependency checks to allow either 1) Known-clients file and/or CA clients(default truststore) 2) Custom truststore

File: besu/src/main/java/org/hyperledger/besu/cli/options/JsonRpcHttpOptions.java [hunk 3]
```diff
@@ -392,12 +409,27 @@ private void validateTls(final CommandLine commandLine) {
...
+          && rpcHttpTlsTruststoreFile == null) {
+        throw new CommandLine.ParameterException(
+            commandLine,
+            "Known-clients file or Truststore file must be specified or CA clients must be enabled when TLS client authentication is enabled for JSON-RPC HTTP endpoint");
+      }
+
+      if (rpcHttpTlsTruststoreFile != null && rpcHttpTlsTruststorePasswordFile == null) {
+        throw new CommandLine.ParameterException(
+            commandLine,
+            "Truststore password file is required when truststore file is specified for JSON-RPC HTTP endpoint");
```
Comment (macfarla) [inline besu/src/main/java/org/hyperledger/besu/cli/options/JsonRpcHttpOptions.java:424]: would be helpful to suggest the CLI option needed here

File: besu/src/main/java/org/hyperledger/besu/cli/options/JsonRpcHttpOptions.java [hunk 4]
```diff
@@ -392,12 +409,27 @@ private void validateTls(final CommandLine commandLine) {
...
-      throw new CommandLine.ParameterException(
-          commandLine,
-          "Known-clients file must be specified or CA clients must be enabled when TLS client authentication is enabled for JSON-RPC HTTP endpoint");
+    if (isRpcHttpTlsClientAuthEnabled) {
+      if (!isRpcHttpTlsCAClientsEnabled
+          && rpcHttpTlsKnownClientsFile == null
+          && rpcHttpTlsTruststoreFile == null) {
+        throw new CommandLine.ParameterException(
+            commandLine,
+            "Known-clients file or Truststore file must be specified or CA clients must be enabled when TLS client authentication is enabled for JSON-RPC HTTP endpoint");
```
Comment (macfarla) [inline besu/src/main/java/org/hyperledger/besu/cli/options/JsonRpcHttpOptions.java:418]: would be helpful to suggest the CLI option needed here

**Review by macfarla — COMMENTED** [review]
with the error messages, it's helpful for the user to indicate exactly which CLI option is missing or required. Otherwise i think it's looking good

**Comment by macfarla** [comment]
There is a unit test failing. You can run these tests locally, this will benefit your time and reviewers. You'll get feedback waaaayy faster. converting to draft since it's not ready to review

```
JsonRpcHttpOptionsTest > rpcHttpTlsClientAuthWithTrustStoreAndKnownClientsFileReportsError() FAILED
    java.lang.AssertionError at JsonRpcHttpOptionsTest.java:508
```

**Comment by pullurib** [comment]
> There is a unit test failing. You can run these tests locally, this will benefit your time and reviewers. You'll get feedback waaaayy faster. converting to draft since it's not ready to review
> 
> ```
> JsonRpcHttpOptionsTest > rpcHttpTlsClientAuthWithTrustStoreAndKnownClientsFileReportsError() FAILED
>     java.lang.AssertionError at JsonRpcHttpOptionsTest.java:508
> ```

I agree,  no change is too small to skip running local tests. Fixed and ran the tests locally.

**Review by macfarla — APPROVED** [review]

**Pull request merged**